src/lib.rs

```diff
@@ -24,6 +25,16 @@ pub trait Bitsized {
     const MAX: Self::ArbitraryInt;
 }
 
+/// Internally used marker trait. 
+/// # Safety 
+/// 
+/// Avoid implementing this for your types. Implementing this trait could break invariants.
+pub unsafe trait Filled: Bitsized {}
+unsafe impl<T> Filled for T where T: Bitsized + From<<T as Bitsized>::ArbitraryInt> {}
```

Comment:
Can we seal the trait / make it unimplementable by users?
Afaik as long as Bitsized is public, it should still implement itself automatically for all `FromBits` types the user defines.
https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/
`supertrait` should be enough

If that won't work, then I remember:
https://github.com/google/zerocopy/blob/b5b30d0287734c004bb2c1c97cc53a460443a57f/src/lib.rs#L238
which at least spells out "don't do this" if you try to implement it

Reply:
feel free to correct me but I think this isn't possible. say you've got
```rust
mod private {
    pub trait Sealed {}
}

pub trait Filled: private::Sealed {}
```
but now you're stuck, right? because now your bounds are
```rust
impl<T> Filled for T where T: Bitsized + From<<T as Bitsized>::ArbitraryInt> + private::Sealed {}
```
but you're generating code that impls `Bitsized` for new structs/enums on the user's side. and you can't implement `private::Sealed` over there.

Reply:
also if we want even more `<#ty as Trait>::something()` in our code, we can do 
```rust
pub trait Filled {
    fn assume_filled() {}
}
```

Reply:
Ok I just tried implementing `Filled` in one of the examples and it's rejected anyways, so we can remove `unsafe` there.
![image](https://github.com/hecatia-elegua/bilge/assets/108802164/2aefd127-91df-45e2-a2fe-c25002e856d6)

(and you're right, supertrait seal would not work, but method signature seal would have)